### ISSUE TITLE
gateleen-expansion: Configuration broken

### ISSUE THREAD

**Issue by mcweba** [swisspush/gateleen#139]
According to the documentation, the gateleen-expansion module uses the following configurations:

- max.recursion.depth
- max.expansion.subrequests

The _max.expansion.subrequests_ property works as expected by returning a **400 Bad Request** when the limit of sub requests is exceeded.

However, the _max.recursion.depth_ property has a not very obvious behaviour: The configured value does not have any effect, unless the expand parameter is not a valid number.

**Example**: _max.recursion.depth_ is configured to the value 2. When making the following request

```
GET http://localhost:7012/gateleen/some_resources?expand=4xyz
```
the expand parameter (internally) changed to the value 2, resulting in a regular _expand=2_ expansion request. This is not so obvious, since the intended expand parameter was obviously 4.

Having the same configured _max.recursion.depth_ value of 2 when making the request

```
GET http://localhost:7012/gateleen/some_resources?expand=4
```
results in a regular _expand=4_ expansion request returning the whole 4 levels.

### max.expansion.subrequests
This configuration property will not be changed. Whenever the configured amount of sub requests is exceeded, the expansion request will be answered with **400 Bad Request**.

### max.expansion.level.soft
This property replaces the _max.recursion.depth_ property. When the provided expand level is greater than the _max.expansion.level.soft_ value, the expand parameter is changed (reduced) to the _max.expansion.level.soft_ value and the expanded resources will be returned (with status 200 OK). Additionally, a warning is logged because the returned expanded resources are not in the intended expansion level.

### max.expansion.level.hard
This configuration property defines the maximum allowed expand level. When the provided expand level is greater thant the _max.expansion.level.hard_ value, a **400 Bad Request** is returned. This check can be made right at the entrance of the expansion module before actually expanding anything, so no load for any sub requests will be generated.

## Reject invalid expand parameter value requests
The mentioned expand request

```
GET http://localhost:7012/gateleen/some_resources?expand=4xyz
```
with the _expand=4xyz_ parameter should not be magically changed to something else. Since it's obviously a bad request, it should be answered with **400 Bad Request**.

**Comment by lbovet** [comment]
For simplicity, I would just keep max.recursion.depth and apply the behaviour you describe in max.epansion.level.hard. I don't find nice to silently change the level set by the user.

**Comment by mcweba** [comment]
The current implementation of _max.recursion.depth_ does silently change the level set by the user but only when the provided expand parameter is not a valid number. When the provided expand parameter is valid but greater than _max.recursion.depth_, the _max.recursion.depth_ value does not have any effect at all.

**Comment by mcweba** [comment]
Implemented in [v1.0.50](https://github.com/swisspush/gateleen/releases/tag/v1.0.50)
